### PR TITLE
Support @Consume 

### LINKED ISSUE #2580: Support @Consume
`@org.apache.camel.Consume` https://camel.apache.org/manual/latest/pojo-consuming.html

### PR DESCRIPTION
**Pull request by ppalaga**
Fix #2580

### CODE REVIEW

File: extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/ConsumeProcessor.java [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+ */
+public class ConsumeProcessor {
+
+    private static final DotName CONSUME_ANNOTATION = DotName.createSimple(Consume.class.getName());
+    private static final DotName NAMED_ANNOTATION = DotName.createSimple(Named.class.getName());
+    /**
+     * Based on https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#builtin_scopes
+     * the list is not 100% complete, but hopefully it will suffice for our purposes
+     */
+    public static final Set<DotName> BEAN_DEFINIG_ANNOTATIONS = new HashSet<DotName>(Arrays.asList(
```
Comment (jamesnetherton) [inline extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/ConsumeProcessor.java:72]: Typo `DEFINIG`.

Comment (ppalaga) [reply extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/ConsumeProcessor.java:72]: Thanks, let me fix it

**Comment by ppalaga** [comment]
Ups, I have pushed bad HEAD, which closed the PR inadvertently. Let me open a new PR.

**Comment by ppalaga** [comment]
See https://github.com/apache/camel-quarkus/pull/2588

**Pull request closed without merging**